mas-foundation/src/main/java/com/ca/mas/foundation/MASUser.java

```
@@ -306,9 +301,73 @@ public void onError(Throwable e) {
             @Override
             public void logout(final MASCallback<Void> callback) {
                 current = null;
-                new LogoutAsyncTask().execute(callback);
+                logout(true, callback);
             }
 
+            /**
+             * <b>Description:</b> Logout from the server.
+             * @param callback MASCallback
+             * @param force by default true it will clean the localstorage
+             */
+            public void logout(final boolean force, final MASCallback<Void> callback) {
+                current = null;
+
+                final TokenManager tokenManager = StorageProvider.getInstance().getTokenManager();
+                final ClientCredentialContainer clientCredentialContainer = StorageProvider.getInstance().getClientCredentialContainer();
+                IdToken idToken = tokenManager.getIdToken();
+                String clientId = clientCredentialContainer.getClientId();
+                String clientSecret = clientCredentialContainer.getClientSecret();
+
+                List<Pair<String, String>> form = new ArrayList<>();
+                form.add(new Pair<>(OAuthClient.ID_TOKEN, idToken.getValue()));
+                form.add(new Pair<>(OAuthClient.ID_TOKEN_TYPE, idToken.getType()));
+                form.add(new Pair<>(OAuthClient.LOGOUT_APPS, Boolean.toString(true)));
+
+                String endpointPatht = MASConfiguration.getCurrentConfiguration().getEndpointPath(MobileSsoConfig.PROP_TOKEN_URL_SUFFIX_RESOURCE_OWNER_LOGOUT);
+                URI uri = null;
+                try {
+                    uri = new URI(endpointPatht);
+                } catch (URISyntaxException e) {
+                    e.printStackTrace();
```

Rethrow runtime exception instead of ignoring it.